### PR TITLE
 chore: use SCLs Nginx base image 

### PR DESCRIPTION
**Pull request by zregvart**
**HEADS UP** @syndesisio/ui-api @cunningt @djcoleman @orpiske

Switches the use of the Syndesis UI Docker base image from CentOS and Nginx installed from nginx.org RPM repository to SCLs Nginx[1].

Since SCLs Nginx is a S2I builder image the Dockerfile was modified to copy the distribution files and configuration to `/tmp/src` and run `assemble` as an S2I build would do.

`docker.base-image` Maven property can be used to change to a different SCLs Nginx base image (say `rhscl/nginx-114-rhel7` for RHEL 7 based base image).

[1] https://github.com/sclorg/nginx-container/blob/master/1.14/README.md

### CODE REVIEW

**Comment by paoloantinori** [comment]
Merging of the `syndesis-react` is in progress. 
We have to remember to replicate this activity on those files as well.

**Comment by heiko-braun** [comment]
Please hold it until master is free for 7.5 development. We are not going to include it 7.4

**Comment by riccardo-forina** [comment]
Master is now about 7.5, but this requires a rebase.

**Comment by christophd** [comment]
@zregvart @riccardo-forina is this still on hold?

**Comment by zregvart** [comment]
I think we can merge this, fancy doing a review @christophd?

File: app/ui-angular/pom.xml [hunk 1]
```diff
@@ -35,6 +35,7 @@
     <yarn-install-args />
     <yarn-verbose />
     <npm-verbose />
+    <docker.base-image>centos/nginx-114-centos7</docker.base-image>
```
Comment (christophd) [inline app/ui-angular/pom.xml]: Mixing `.` and `-` in property names is something I would not prefer doing. Just a matter of personal taste I guess

Comment (zregvart) [reply app/ui-angular/pom.xml]: Not actually sure that property is needed, I wonder why I added that?

Comment (zregvart) [reply app/ui-angular/pom.xml]: Turns out that the fabric8-maven-plugin requires this property to build the image using S2I on OpenShift. Not sure where is this documented or how my past self knew about this. The error when the property is removed is:

```
[ERROR] Failed to execute goal io.fabric8:fabric8-maven-plugin:3.5.38:build (exec) on project ui: Failed to execute the build: Unable to build the image using the OpenShift build service: OpenShift Build syndesis-ui-legacy-s2i-2: Failed: PullBuilderImageFailed -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[INFO] F8: BuildWatch: Received event MODIFIED , build status: BuildStatus(cancelled=null, completionTimestamp=null, config=ObjectReference(apiVersion=null, fieldPath=null, kind=BuildConfig, name=syndesis-ui-legacy-s2i, namespace=master, resourceVersion=null, uid=null, additionalProperties={}), duration=null, logSnippet=null, message=Failed pulling builder image., output=BuildStatusOutput(to=null, additionalProperties={}), outputDockerImageReference=172.30.1.1:5000/master/syndesis-ui-legacy:latest, phase=Failed, reason=PullBuilderImageFailed, stages=[StageInfo(durationMilliseconds=null, name=PullImages, startTime=2019-07-02T09:07:26Z, steps=[StepInfo(durationMilliseconds=null, name=PullBaseImage, startTime=2019-07-02T09:07:26Z, additionalProperties={})], additionalProperties={})], startTimestamp=2019-07-02T09:06:39Z, additionalProperties={})
[INFO] F8: Build syndesis-ui-legacy-s2i-2 status: Failed
[WARNING] Ignoring error for already closed/closing connection
```

**Comment by zregvart** [comment]
I think I need to make the same change in the `ui-react`, I thought that was done already.

File: app/ui-angular/pom.xml (hunk 1 above)
Comment (zregvart) [reply app/ui-angular/pom.xml]: Aah, found it in the `Dockerfile`.

**Pull request merged**